packages/app/cypress/e2e/support/execute-spec.ts

```diff
@@ -25,11 +25,17 @@ declare global {
 }
 
 export const waitForSpecToFinish = (expectedResults, timeout?: number) => {
-  // First ensure the test is loaded
-  cy.get('.passed > .num').should('exist')
-  cy.get('.failed > .num').should('exist')
+  // Check if we're in studio mode using URL hash parameters (same method as studio store)
+  cy.window().then((win) => {
+    const studioSingleTestActive = win.state?.studioSingleTestActive
```

Comment:
If this is becoming too much of an issue, just pass in `isStudioMode` similar to what you were doing before.